src/WpfMath/TexFormulaParser.cs

```diff
@@ -393,20 +395,58 @@ private TexFormula ReadScript(TexFormula formula, SourceSpan value, ref int posi
 
                     source = value.Segment(start, sqrtEnd - start);
                     return new Radical(source, sqrtFormula.RootAtom, degreeFormula?.RootAtom);
-
-                case "color":
+                    }
+                case "fgcolor":
                     {
+                        //Command to change the foreground color 
                         var colorName = ReadGroup(formula, value, ref position, leftGroupChar, rightGroupChar);
-                        var remainingString = value.Segment(position);
+
+                        string remainingString = value.Substring(position);
```

Comment:
I haven't been able to work with the SourceSpan and Result structure because I'm stilk using VS 15.6.7 and they through a lot of errors when I try to build it.

Reply:
Well that's a pity. I'm not sure how to proceed. In this project (and generally in open-source) I tend to touch all the shiny new things. because it's interesting and educational experience for all the members involved. "Having fun" is the main driving force for my open-source contributions (while "creating good software packages and providing them to the users" is also a goal, it's not the driving force). Thus I don't think that falling back to ol' good C# 7.0 is an option for us.

Although I fully understand that there're users whose work environment enforces some strict policies and that updating the Visual Studio isn't an option in such environments.

I'm not sure how to proceed. Could you maybe set up your other machine with a more recent version of Visual Studio Community edition (it is enough for developing open source software)? Or even try to set it up using the Visual Studio Build Tools package and VSCode? _(Not sure if it includes the WPF build tools though, likely it isn't.)_

Reply:
Also you should look at the build server checks at the bottom of that page: e.g. [here](https://ci.appveyor.com/project/ForNeVeR/wpf-math/build/1.0.453) it complains about SourceSpan.